src/js/components/search/visualizations/geo/map/MapBox.jsx

```diff
@@ -56,6 +68,67 @@ export default class MapBox extends React.Component {
         });
     }
 
+    goDirection(direction, map) {
+        switch (direction) {
+            case 'left':
+                map.panBy([-delta, 0]);
+                break;
+            case 'right':
+                map.panBy([delta, 0]);
+                break;
+            case 'up':
+                map.panBy([0, -delta]);
+                break;
+            case 'down':
+                map.panBy([0, delta]);
+                break;
+            default:
+                break;
+        }
+    }
+
+    buttonListen(button, map) {
+        this.persist = setInterval(() => {
```

Comment:
Why are the button events timers? Is it so you can hold them down and constantly pan?
Also, if they must be timers, can you ensure they check for existing timers and cancel them? If a user brings up 3D Touch or a context menu, the map begins to infinitely scroll.